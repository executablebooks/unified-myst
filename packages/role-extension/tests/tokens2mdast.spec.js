import { fromMarkdown } from 'mdast-util-from-markdown'

import { mystRoleMmarkExt, mystRoleMdastExt } from '../src/index.js'

const fixtures = ['{name}`content`', '{name}`multi\nline`\n']

describe('Convert source to MDAST', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystRoleMmarkExt],
            mdastExtensions: [mystRoleMdastExt],
        }
        const mdast = fromMarkdown(content, options)
        expect(mdast).toMatchSnapshot()
    })
})
