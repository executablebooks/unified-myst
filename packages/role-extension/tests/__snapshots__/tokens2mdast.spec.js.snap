// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to MDAST "{name}\`content\`" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "content": "content",
          "name": "name",
          "position": Object {
            "end": Object {
              "column": 16,
              "line": 1,
              "offset": 15,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "mystRole",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 16,
          "line": 1,
          "offset": 15,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 16,
      "line": 1,
      "offset": 15,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "{name}\`multi\\nline\`\\n" 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "content": "multi
line",
          "name": "name",
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 2,
              "offset": 18,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "mystRole",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 6,
          "line": 2,
          "offset": 18,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 3,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
